ui/src/app/pages/homepage/recent-resources.tsx

```diff
@@ -1,99 +1,137 @@
-import {WorkspaceResource} from 'generated/fetch';
-import {withContentRect} from 'react-measure';
-
-export const RecentResources = (fp.flow as any)(
-  withContentRect('client'),
-)(class extends React.Component<{
-  measureRef: React.Ref<any>,
-  contentRect: {client: {width: number}},
-  dark: boolean
-}, {
-  loading: boolean,
-  offset: number,
-  resources: WorkspaceResource[],
-  existingCohortName: string[],
-  existingConceptName: string[],
-  existingNotebookName: string[]
-}> {
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: false,
-      resources: [],
-      offset: 0,
-      existingCohortName: [],
-      existingConceptName: [],
-      existingNotebookName: []};
-  }
+import {userMetricsApi, workspacesApi} from 'app/services/swagger-fetch-clients';
+import {formatWorkspaceResourceDisplayDate, getCdrVersion, reactStyles, withCdrVersions} from 'app/utils';
+import {navigateAndPreventDefaultIfNoKeysPressed} from 'app/utils/navigation';
+import {getDisplayName} from 'app/utils/resources';
+import {CdrVersionListResponse, Workspace, WorkspaceResource, WorkspaceResourceResponse} from 'generated/fetch';
 
-  componentDidMount() {
-    this.loadResources();
+const styles = reactStyles({
+  column: {
+    textAlign: 'left',
+  },
+  typeColumn: {
+    textAlign: 'left',
+    width: '130px',
+  },
+  menu: {
+    width: '30px',
+  },
+  navigation: {
+    fontFamily: 'Montserrat',
+    fontSize: '14px',
+    letterSpacing: 0,
+    lineHeight: '22px',
   }
+});
 
-  async loadResources() {
-    try {
-      this.setState({loading: true});
-      const resources = await userMetricsApi().getUserRecentResources();
-      this.setState({resources});
-    } catch (error) {
-      console.error(error);
-    } finally {
-      this.setState({loading: false});
-    }
-  }
+const WorkspaceNavigation = (props: {workspace: Workspace, style?: CSSProperties}) => {
+  const {workspace: {name, namespace, id}, style} = props;
+  const url = `/workspaces/${namespace}/${id}/data`;
+
+  return <Clickable>
+    <a data-test-id='workspace-navigation'
+       style={style}
+       href={url}
+       onClick={e => navigateAndPreventDefaultIfNoKeysPressed(e, url)}>
+      {name}
+    </a>
+  </Clickable>;
+};
+
+interface TableData {
+  menu: JSX.Element;
+  resourceType: JSX.Element;
+  resourceName: JSX.Element;
+  workspaceName: JSX.Element;
+  formattedLastModified: string;
+  cdrVersionName: string;
+}
+
+const RecentResources = fp.flow(withCdrVersions())((props: {cdrVersionListResponse: CdrVersionListResponse}) => {
+  const [loading, setLoading] = useState(true);
+  const [resources, setResources] = useState<WorkspaceResourceResponse>();
+  const [wsMap, setWorkspaceMap] = useState<Map<string, Workspace>>();
+  const [tableData, setTableData] = useState<TableData[]>();
+
+  const loadResources = () => {
+    setLoading(true);
+    return userMetricsApi().getUserRecentResources()
```

Comment:
ooh, when did this get renamed? i like this name

Reply:
I'm guessing a while ago, because this is the only name I'm familiar with